dftimewolf/metawolf/utils.py

```diff
@@ -0,0 +1,361 @@
+
+    Returns:
+      bool: True if the processes should keep running in the background.
+    """
+    still_running = False
+    for metawolf_process in processes:
+      if metawolf_process.Poll() is None:
+        still_running = True
+        break
+    if still_running:
+      value = input('Metawolf still has running processes. Would you like to '
+                    'keep them running in the background [Yn]? ') or 'y'
+      q = self.Str2Bool(str(value))
+      while q not in [False, True]:
+        value = input('[Yn]? ') or 'y'
+        q = self.Str2Bool(str(value))
+      return q
+    return True
+
+  def PrepareDFTimewolfCommand(
+      self,
+      recipe: str,
+      session_settables: Dict[str, session.SessionSettable]
+  ) -> List[str]:
+    """Builds a cli command from current session parameters.
+
+    Args:
+      recipe (str): The name of the recipe to run.
+      session_settables (Dict[str, SessionSettable]): The dictionary holding the
+          session settables to use to build the command.
+
+    Returns:
+      List[str]: A list that contains the CLI command elements based on the
+          current recipe and settables.
+    """
+    cmd_components = {}
+    for _, settable in session_settables.items():
+      if settable.name == SESSION_ID_SETTABLE:
+        continue
+      if not settable.IsOptional():
+        value = settable.GetValue()
+        if value is None:
+          return []
+        cmd_components[settable.name] = '{0!s}'.format(value)
+      else:
+        value = settable.GetValue()
+        if value:
+          cmd_components[settable.name] = '--{0:s}={1!s}'.format(
+              settable.name, value)
+
+    # We need to order the arguments as they are defined in the recipe
+    cmd = [DFTIMEWOLF, recipe]
+    for name, _, _ in self.recipe_manager.Recipes()[recipe].args:
+      if name.startswith('--'):
+        name = name.replace('--', '')
+      if name in cmd_components:
+        cmd.append(cmd_components[name])
+    return cmd
+
+  def GetRecipes(self) -> Dict[str, str]:
+    """Return available DFTimewolf recipes.
+
+    Returns:
+      Dict[str, str]: A dictionary that maps recipe names to their short
+          description.
+    """
+    recipes = {}
+    for recipe in self.recipe_manager.GetRecipes():
+      short_desc = ' '.join(recipe.GetHelpString().split())
+      short_desc = short_desc.replace(recipe.name, '').strip()
+      recipes[recipe.name] = short_desc
+    return recipes
+
+  def GetRecipe(self, recipe_name: str) -> Optional[resources.Recipe]:
+    """Return a DFTimewolf Recipe object.
+
+    Args:
+      recipe_name (str): The name of the recipe to return.
+
+    Returns:
+      Recipe: The recipe object if recipe_name was found, or None.
+    """
+    return self.Recipes().get(recipe_name)
+
+  def Recipes(self) -> Dict[str, resources.Recipe]:
+    """Return a dictionary that maps recipe names to Recipe objects."""
+    return self.recipe_manager.Recipes()
```

Comment:
Are these really necessary? They're just wrappers around a no-to-much-longer function calls?

Reply:
Removed